src/services/codefixes/helpers.ts

```diff
@@ -53,108 +94,148 @@ namespace ts.codefix {
+                    const signature = signatures[0];
+                    return signatureToMethodDeclaration(signature, enclosingDeclaration, createStubbedMethodBody());
                 }
 
-                let result = "";
+                const signatureDeclarations = [];
                 for (let i = 0; i < signatures.length; i++) {
-                    const sigString = checker.signatureToString(signatures[i], enclosingDeclaration, TypeFormatFlags.SuppressAnyReturnType, SignatureKind.Call);
-                    result += `${visibility}${name}${sigString};${newlineChar}`;
+                    const signature = signatures[i];
+                    const methodDeclaration = signatureToMethodDeclaration(signature, enclosingDeclaration);
+                    if (methodDeclaration) {
+                        signatureDeclarations.push(methodDeclaration);
+                    }
                 }
 
-                // If there is a declaration with a body, it is the last declaration,
-                // and it isn't caught by `getSignaturesOfType`.
-                let bodySig: Signature | undefined = undefined;
                 if (declarations.length > signatures.length) {
-                    bodySig = checker.getSignatureFromDeclaration(declarations[declarations.length - 1] as SignatureDeclaration);
+                    const signature = checker.getSignatureFromDeclaration(declarations[declarations.length - 1] as SignatureDeclaration);
+                    const methodDeclaration = signatureToMethodDeclaration(signature, enclosingDeclaration, createStubbedMethodBody());
+                    if (methodDeclaration) {
+                        signatureDeclarations.push(methodDeclaration);
+                    }
                 }
                 else {
                     Debug.assert(declarations.length === signatures.length);
-                    bodySig = createBodySignatureWithAnyTypes(signatures, enclosingDeclaration, checker);
+                    const methodImplementingSignatures = createMethodImplementingSignatures(signatures, name, optional, modifiers);
+                    signatureDeclarations.push(methodImplementingSignatures);
                 }
-                const sigString = checker.signatureToString(bodySig, enclosingDeclaration, TypeFormatFlags.SuppressAnyReturnType, SignatureKind.Call);
-                result += getStubbedMethod(visibility, name, sigString, newlineChar);
-
-                return result;
+                return signatureDeclarations;
             default:
-                return "";
+                return undefined;
+        }
+
+        function signatureToMethodDeclaration(signature: Signature, enclosingDeclaration: Node, body?: Block) {
+            const signatureDeclaration = <MethodDeclaration>checker.signatureToSignatureDeclaration(signature, SyntaxKind.MethodDeclaration, enclosingDeclaration);
+            if (signatureDeclaration) {

```

Comment:
May introduce new hidden class? Ensure `decorators` is assigned to `undefined` either here or in `signatureToSignatureDeclaration`.